### ISSUE TITLE
WARNING: OOM in all workers, skipping update

### ISSUE THREAD

**Issue by Hasan0ff** [modernmt/modernmt#466]
```
| WARNING: OOM in all workers, skipping update
| WARNING: ran out of memory with exception: CUDA out of memory. Tried to allocate 100.12 MiB (GPU 0; 1.95 GiB total capacity; 1.30 GiB already allocated; 60.31 MiB free; 112.66 MiB cached);
 Skipping batch
| WARNING: OOM in all workers, skipping update
| epoch 016:   1300 / 3369 loss=5.832, nll_loss=4.468, ppl=22.14, wps=274, ups=0, wpb=581.872, bsz=8.167, num_updates=8161, lr=7.00097e-05, gnorm=2.718, clip=0.000, oom=0.843, wall=16208, train_w$
| WARNING: ran out of memory with exception: CUDA out of memory. Tried to allocate 107.88 MiB (GPU 0; 1.95 GiB total capacity; 1.27 GiB already allocated; 100.31 MiB free; 108.49 MiB cached);
 Skipping batch
| WARNING: OOM in all workers, skipping update
| WARNING: ran out of memory with exception: CUDA out of memory. Tried to allocate 114.75 MiB (GPU 0; 1.95 GiB total capacity; 1.30 GiB already allocated; 104.31 MiB free; 69.21 MiB cached);
 Skipping batch
```

I checked the training log after 12 hours of training and it's full of remarks such as these. Is it possible that lack of GPU RAM is directly related to the bad translation i get? Does this log reference that the training isn't properly done?

**Comment by Hasan0ff** [comment]
```
root@s4pc-s17:~/modernmt# ./mmt translate "你好"
Recalling that, in accordance with the Charter of Afghanistan,
root@s4pc-s17:~/modernmt# ./mmt translate "谢谢"
15. Decides that, in accordance with the provisions of its resolution 973 (X) of 15 December 1955, there
root@s4pc-s17:~/modernmt#
```
First one is Ni Hao which is Hello.. Second is Shie Shie which is Thank you :|

**Comment by Hasan0ff** [comment]
The third problem that could end in bad translation could be result of this: 

https://i.ibb.co/ynQVvTV/Screen-Shot-2019-05-01-at-4-40-32-AM.png

This is the screenshot of opus website from where i downloaded the corpora. For test, I downloaded zip file under Moses section belonging to Moses format.

( http://opus.nlpl.eu/download.php?f=UN/v20090831/moses/en-zh.txt.zip )

Afterwards, I unzipped the file in examples/data/train/ and began the training. Is it possible that I have done it wrongly by adding the Moses files which I think aren't aligned properly.
